app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationAnimationActivity.kt

```diff
@@ -0,0 +1,162 @@
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.databinding.ItemCalloutViewBinding
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.turf.TurfConstants.UNIT_DEFAULT
+import com.mapbox.turf.TurfMeasurement
+import kotlin.math.roundToLong
+
+/**
+ * Example how to animate a view annotations on top of a route.
+ */
+class ViewAnnotationAnimationActivity : AppCompatActivity() {
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private lateinit var annotationView: View
+  private lateinit var textView: TextView
+  private var routeCoordinateList = mutableListOf<Point>()
+  private var currentRouteCoordinate = 0
+  private var totalLength = 0.0
+  private var currentAnimator: Animator? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val mapView = MapView(this)
+    setContentView(mapView)
+
+    // load feature collection from assets file
+    val featureCollection = FeatureCollection.fromJson(
+      AnnotationUtils.loadStringFromAssets(
+        this, ROUTE_FILE_NAME
+      )!!
+    )
+
+    // calculate the route length, get coordinates from route geometry
+    val lineString = featureCollection.features()!![0].geometry() as LineString
+    routeCoordinateList = lineString.coordinates()
+    totalLength = TurfMeasurement.length(lineString, UNIT_DEFAULT)
+
+    // initialize the mapview
+    mapView.getMapboxMap().apply {
+      loadStyle(
+        style(Style.LIGHT) {
+          // source for displaying the route
+          +geoJsonSource(SOURCE_ID) {
+            featureCollection(featureCollection)
+          }
+          // layer for displaying the route
+          +lineLayer(LINE_ID, SOURCE_ID) {
+            lineColor(Color.parseColor(COLOR_PINK_HEX))
+            lineWidth(4.0)
+          }
+        }
+      ) {
+        // center camera around SF airport
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(-122.3915, 37.6177))
+            .zoom(11.0)
+            .build()
+        )
+        // get initial point
+        val initialPoint = routeCoordinateList[0]
+
+        // create view annotation callout item
+        viewAnnotationManager = mapView.viewAnnotationManager
+        annotationView = viewAnnotationManager.addViewAnnotation(
+          R.layout.item_callout_view,
+          ViewAnnotationOptions.Builder()
+            .geometry(initialPoint)
+            .build()
+        )
+
+        ItemCalloutViewBinding.bind(annotationView).apply {
+          textView = textNativeView
+        }
+
+        animateNextStep()
+      }
+    }
+  }
+
+  override fun onStop() {
+    super.onStop()
+    currentAnimator?.cancel()
+  }
+
+  private fun animateNextStep() {
+    currentRouteCoordinate %= (routeCoordinateList.size - 1)
```

Comment:
for me `currentRouteCoordinate` should store coordinate as an object or pair.
so, maybe rename it.

```suggestion
    currentRouteCoordinateIndex %= (routeCoordinateList.size - 1)
```